test/metabase/api/search_test.clj

```diff
@@ -336,37 +340,42 @@
       (is (= #{} (get-available-models :q "noresults"))))))
 
 (deftest available-models-test
-  (let [search-term "query-model-set"]
-    (with-search-items-in-root-collection search-term
-      (testing "should returns a list of models that search result will return"
-        (is (= #{"dashboard" "table" "dataset" "segment" "collection" "database" "action" "metric" "card"}
-               (get-available-models :q search-term))))
-      (testing "return a subset of model for created-by filter"
-        (is (= #{"dashboard" "dataset" "card" "metric" "action"}
-               (get-available-models :q search-term :created_by (mt/user->id :rasta)))))
-      (testing "return a subset of model for verified filter"
-        (t2.with-temp/with-temp
-          [:model/Card       {v-card-id :id}   {:name (format "%s Verified Card" search-term)}
-           :model/Card       {v-model-id :id}  {:name (format "%s Verified Model" search-term) :type :model}
-           :model/Card       {v-metric-id :id} {:name (format "%s Verified Metric" search-term) :type :metric}
-           :model/Collection {_v-coll-id :id}  {:name (format "%s Verified Collection" search-term) :authority_level "official"}]
-          (testing "when has both :content-verification features"
-            (mt/with-premium-features #{:content-verification}
-              (mt/with-verified-cards! [v-card-id v-model-id v-metric-id]
-                (is (= #{"card" "dataset" "metric"}
-                       (get-available-models :q search-term :verified true))))))
-          (testing "when has :content-verification feature only"
-            (mt/with-premium-features #{:content-verification}
-              (mt/with-verified-cards! [v-card-id]
-                (is (= #{"card"}
-                       (get-available-models :q search-term :verified true))))))))
-      (testing "return a subset of model for created_at filter"
-        (is (= #{"dashboard" "table" "dataset" "collection" "database" "action" "card" "metric"}
-               (get-available-models :q search-term :created_at "today"))))
-
-      (testing "return a subset of model for search_native_query filter"
-        (is (= #{"dataset" "action" "card" "metric"}
-               (get-available-models :q search-term :search_native_query true)))))))
+  ;; Porting these tests over earlier
+  (doseq [engine ["in-place" "fulltext"]]
+    (let [search-term "query-model-set"
+          get-available-models #(apply get-available-models :search_engine engine %&)]
+      (with-search-items-in-root-collection search-term
+        (testing "should returns a list of models that search result will return"
+          (is (= #{"dashboard" "table" "dataset" "segment" "collection" "database" "action" "metric" "card"}
+                 (get-available-models)))
```

Comment:
Even the API tests were not exercising this "no query string" edge-case, so even if we'd finish porting over the whole namespace as planned it would not have caught it.

I don't think it even makes sense it many of the context to be skipping the query string - it's pretty weird in the `search-app` that we'll show models with no results for the current search, without even a search box where they can change that query.